### PR TITLE
DS-2965 Update UDF get_earliest_value()

### PR DESCRIPTION
**Pull request by lucia-vargas-a**
This PR updates the UDF `get_earliest_value()` to facilitate the implementation of the new profiles re-definition. 
Jira [DS-2965](https://mozilla-hub.atlassian.net/browse/DS-2965).

It also reduce the lines and complexity required to retrieve earliest ping.

> Note for the reviewer. Many files are reformatted to pass the CI. The files directly affected by the change are under fenix_derived/firefox_android_clients_v1 and mozfun/norm/get_earliest_value. 

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

[DS-2965]: https://mozilla-hub.atlassian.net/browse/DS-2965?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: sql/moz-fx-data-marketing-prod/ga_derived/downloads_with_attribution_v2/checks.sql [hunk 1]
```diff
@@ -1,2 +1,12 @@
-ASSERT((SELECT COUNT(*) FROM `{{project_id}}.{{dataset_id}}.{{table_name}}` WHERE download_date = @download_date) > 250000)
-AS 'ETL Data Check Failed: Table {{project_id}}.{{dataset_id}}.{{table_name}} contains less than 250,000 rows for date: {{ download_date }}.'
\ No newline at end of file
+ASSERT(
```
Comment (lucia-vargas-a) [inline sql/moz-fx-data-marketing-prod/ga_derived/downloads_with_attribution_v2/checks.sql:1]: Format changes only.

File: sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/schema.yaml [hunk 2]
```diff
@@ -76,6 +76,42 @@ fields:
   mode: NULLABLE
   name: install_source
   type: STRING
+- description: Last reported client locale.
```
Comment (lucia-vargas-a) [inline sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/schema.yaml:79]: Moving the schema to match the init file.

File: sql/moz-fx-data-shared-prod/telemetry_derived/clients_histogram_aggregates_v2/query.sql [hunk 3]
```diff
@@ -1,5 +1,6 @@
 CREATE TEMP FUNCTION udf_use_old_data(old_aggs ANY TYPE)
-  RETURNS ARRAY<STRUCT<
+RETURNS ARRAY<
```
Comment (lucia-vargas-a) [inline sql/moz-fx-data-shared-prod/telemetry_derived/clients_histogram_aggregates_v2/query.sql:2]: Reformatting

File: sql/moz-fx-data-marketing-prod/ga_derived/downloads_with_attribution_v2/checks.sql (hunk 1 above)
Comment (gleonard-m) [reply sql/moz-fx-data-marketing-prod/ga_derived/downloads_with_attribution_v2/checks.sql:1]: Curious - are these format changes automated or manual?

File: sql/mozfun/norm/get_earliest_value/metadata.yaml [hunk 4]
```diff
@@ -1,9 +1,9 @@
 friendly_name: Get Earliest Value
 description: |
-  This UDF returns the earliest not-null value from a list of values
-  and their corresponding timestamp.
+  This UDF returns the earliest not-null value pair and datetime
```
Comment (gleonard-m) [inline sql/mozfun/norm/get_earliest_value/metadata.yaml:3]: Maybe add that the pair includes the source of the value?

**Review by gleonard-m — APPROVED** [review]

**Review by gleonard-m — COMMENTED** [review]
Updates needed to other failing tests

File: sql/moz-fx-data-marketing-prod/ga_derived/downloads_with_attribution_v2/checks.sql (hunk 1 above)
Comment (lucia-vargas-a) [reply sql/moz-fx-data-marketing-prod/ga_derived/downloads_with_attribution_v2/checks.sql:1]: They're automated in the CI. Locally I run them using ./bqetl format.
E.g. to check all the query files `./bqetl format $(git ls-tree -d HEAD --name-only)`

Comment (gleonard-m) [reply sql/moz-fx-data-marketing-prod/ga_derived/downloads_with_attribution_v2/checks.sql:1]: Interesting that it picked up the checks.sql not just query.sql files.  One less thing to do to cover the checks.sql files.

**Pull request merged**